nebu/cli/get.py

```diff
@@ -158,88 +181,200 @@ def store_sha1(sha1, write_dir, filename):
+                return None
+
+        def get_legacy_id():
+            try:
+                return metadata['legacy_id']
+            except (TypeError, KeyError):
+                return None
+
+        def get_scoped_directory():
+            is_module = content_filename == 'index.cnxml'
+            is_collection = content_filename == 'collection.xml'
+
+            node_title = node.get('title') or ''
+            index_string = ('{:02d} '.format(index_in_group)
+                            if not is_collection
+                            else '')
+            tree_dirname = f'{index_string}{_safe_name(node_title)}'
+
+            if is_module and not book_tree:
+                return legacy_id
+            if book_tree:
+                return tree_dirname
+            return ''
+
+        tasks = []
+        metadata = await get_metadata()
+        resource_groups = get_resource_groups()
+
+        content_filename = get_content_filename()
+        legacy_id = get_legacy_id()
+
+        scoped_directory = get_scoped_directory()
+        write_dir = write_dir / scoped_directory
+        write_dir.mkdir(parents=True, exist_ok=True)
+
+        if resource_groups is not None:
+            enqueue_content()
+            enqueue_extras()
+        enqueue_children()
+
+        await asyncio.wait(tasks)
+        if pbar is not None:
+            pbar.update(1)
+        return
+
+    async def fetch_resource_node(session,
+                                  resource_url,
+                                  write_dir,
+                                  filename,
+                                  resource_id):
+        async def read_response(response):
+            return await response.read()
+        resp = await do_with_retry_request_while(
+            session=session,
+            url=resource_url,
+            condition=lambda resp: resp.status >= 500,
+            max_retries=4,
+            do=read_response)
         filepath = write_dir / filename
-
-        # core files are XML - this parse/serialize removes numeric entities
-        filepath.write_bytes(etree.tostring(etree.XML(file_resp.content),
+        filepath.write_bytes(resp)
+        store_sha1(resource_id, write_dir, filename)
+
+    async def fetch_content_node(session,
+                                 content_url,
+                                 write_dir,
+                                 filename,
+                                 legacy_id):
+        async def read_response(response):
+            return await response.read()
+        resp = await do_with_retry_request_while(
+            session=session,
+            url=content_url,
+            condition=lambda resp: resp.status >= 500,
+            max_retries=4,
+            do=read_response)
+        filepath = write_dir / filename
+        filepath.write_bytes(etree.tostring(etree.XML(resp),
                                             encoding='utf-8'))
-
-        # Cache the sha1 for this node
```

Comment:
Just curious... It seems like the resources will now be re-downloaded even if they already exist. If so, why? It seems like resources take the most time to download

Reply:
In what case would a resource be re-downloaded. Do we have resources present in more than one module?

Reply:
If a user runs `neb get ...` and then runs the same command again will they re-download all the resources?

Reply:
`neb get` cannot be given an output directory that already exists. I do not know the original reasoning for this, but it seems like a requirement for knowing the resources have already been downloaded.

Reply:
`neb get` must use a directory that does not already exist. I don't know the reasoning for that, but I think that would be an issue for detecting if resources had already been downloaded.